storage: Divide-by-zero panic in unit test run under race

When running:

```
make stressrace PKG=./storage/ STRESSFLAGS='-stderr -maxfails 1 -p 32'
```

...on top of #14815, I pretty quickly hit:

```
--- FAIL: TestScannerTiming (0.11s)
panic: runtime error: integer divide by zero [recovered]
  panic: runtime error: integer divide by zero [recovered]
  panic: runtime error: integer divide by zero

goroutine 9941 [running]:
testing.tRunner.func1(0xc4205141a0)
  /home/alex/go1.8/src/testing/testing.go:622 +0x55f
panic(0x21f4900, 0x321b0e0)
  /home/alex/go1.8/src/runtime/panic.go:489 +0x2f0
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
  /home/alex/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:69 +0x59a
panic(0x21f4900, 0x321b0e0)
  /home/alex/go1.8/src/runtime/panic.go:489 +0x2f0
github.com/cockroachdb/cockroach/pkg/storage.(*replicaScanner).avgScan(0xc4202a2900, 0x0)
  /home/alex/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner.go:165 +0x105
github.com/cockroachdb/cockroach/pkg/storage.TestScannerTiming.func1(0xed07f3906, 0x2dc0894d)
  /home/alex/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner_test.go:256 +0x382
github.com/cockroachdb/cockroach/pkg/util.RetryForDuration(0xa7a358200, 0xc420437f58, 0x0, 0x0)
  /home/alex/go/src/github.com/cockroachdb/cockroach/pkg/util/retry.go:31 +0xdd
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonDepth(0x1, 0x3260fc0, 0xc4205141a0, 0xc420437f58)
  /home/alex/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x60
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x3260fc0, 0xc4205141a0, 0xc420437f58)
  /home/alex/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:35 +0x56
github.com/cockroachdb/cockroach/pkg/storage.TestScannerTiming(0xc4205141a0)
  /home/alex/go/src/github.com/cockroachdb/cockroach/pkg/storage/scanner_test.go:263 +0x175
testing.tRunner(0xc4205141a0, 0x2419258)
  /home/alex/go1.8/src/testing/testing.go:657 +0x108
created by testing.(*T).Run
  /home/alex/go1.8/src/testing/testing.go:697 +0x544
```

Looks like an easy fix and a problem that only affected tests.

--- a/pkg/storage/scanner.go
+++ b/pkg/storage/scanner.go
@@ -162,6 +162,9 @@ func (rs *replicaScanner) GetDisabled() bool {
 func (rs *replicaScanner) avgScan() time.Duration {
 	rs.mu.Lock()
 	defer rs.mu.Unlock()
+	if rs.mu.scanCount == 0 {
+		return 0
+	}
 	return time.Duration(rs.mu.total.Nanoseconds() / rs.mu.scanCount)
 }
 

